Update structure and separate v2/v3 documents

ronaldtse opened the pull request:
Need to fix the build process for:

1. Documents not being able to run in xml2rfc (@opoudjis )
2. Relaton IETF failing to read MN IETF documents (@andrew2net )

Can you help take action? Thanks.

ronaldtse committed:
Update structure and separate v2/v3 documents

ronaldtse commented:
Relaton issue. This might be related to Relaton trying to read non-Metanorma files (the `.rfc.xml` files are RFC XML files, not Metanorma XML files).

```
echo "Running locally..."; bundle exec relaton concatenate \
	  -t "Metanorma-IETF sample documents" \
		-g "Internet Engineering Task Force" \
		documents documents.rxl
Running locally...
[relaton-bib] title type "abbrev" is invalid.
[relaton-bib] Series type is invalid: stream
[relaton-bib] title type "abbrev" is invalid.
[relaton-bib] Series type is invalid: stream
[relaton-bib] Series type is invalid: intended
[relaton-bib] title type "abbrev" is invalid.
[relaton-bib] Series type is invalid: stream
[relaton-bib] Series type is invalid: intended
[relaton-bib] title type "abbrev" is invalid.
[relaton-bib] Series type is invalid: stream
bundler: failed to load command: relaton (/Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/relaton)
NoMethodError: undefined method `sub' for nil:NilClass
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-ietf-0.11.1/lib/relaton_ietf/scrapper.rb:312:in `docids'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-ietf-0.11.1/lib/relaton_ietf/scrapper.rb:57:in `fetch_rfc'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:145:in `block in concatenate_files'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:140:in `map'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:140:in `concatenate_files'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:86:in `bibcollection'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:153:in `concatenate_and_write_to_files'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:19:in `concatenate'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/relaton_file.rb:58:in `concatenate'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli/command.rb:33:in `concatenate'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/lib/relaton/cli.rb:20:in `start'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/relaton-cli-0.8.5/exe/relaton:20:in `<top (required)>'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/relaton:23:in `load'
  /Users/me/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bin/relaton:23:in `<top (required)>'
```

ronaldtse commented:
Generated RFC XML problems:

`antioch.xml`: minor errors?
```
Style Warning: (XML Line 000254): Hanging paragraph in clause
Style Warning: (XML Line 000538): Hanging paragraph in clause
Style Warning: (XML Line 000541): Table should have title
Syntax: (XML Line 000026:52): character content of element "revision-date" invalid; must be a string matching the regular expression "([\+\-]?\d{4})((-?)((0[1-9]|1[0-2])((-?)([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6]))))?"
Syntax: (XML Line 000100:88): attribute "obligation" not allowed here; expected attribute "language", "numbered", "removeInRFC", "script" or "toc"
Syntax: (XML Line 000617:26): element "bibitem" not allowed here; expected the element end-tag or element "clause", "definitions" or "terms"
```

`example.xml`:
```
documents/example.rfc.xml(8): Warning: Setting consensus="true" for IETF STD document (this is not the schema default, but is the only value permitted for this type of document)
documents/example.rfc.xml(11): Warning: Removed a redundant ascii= attribute from <title>
documents/example.rfc.xml(161): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(164): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(8): Warning: Expected a <link> with rel="prev" providing the datatracker url for the origin draft, or alternatively a "docName" attribute on <rfc> from which to construct such a <link> element.
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/boilerplate
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/toc
documents/example.rfc.xml(8): Warning: Invalid document after running preptool.
 Created file documents/example.html
documents/example.rfc.xml(8): Warning: Setting consensus="true" for IETF STD document (this is not the schema default, but is the only value permitted for this type of document)
documents/example.rfc.xml(11): Warning: Removed a redundant ascii= attribute from <title>
documents/example.rfc.xml(161): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(164): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(8): Warning: Expected a <link> with rel="prev" providing the datatracker url for the origin draft, or alternatively a "docName" attribute on <rfc> from which to construct such a <link> element.
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/boilerplate
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/toc
documents/example.rfc.xml(8): Warning: Invalid document after running preptool.
```

`antioch-v2.xml`
```
documents/antioch-v2.rfc.xml(103): Error: Expecting an element , got nothing, at /rfc/middle/section[3]/figure[1]
documents/antioch-v2.rfc.xml(140): Error: Expecting an element , got nothing, at /rfc/middle/section[3]/figure[3]
documents/antioch-v2.rfc.xml(209): Error: Expecting an element , got nothing, at /rfc/middle/section[4]/section[1]/figure[1]
documents/antioch-v2.rfc.xml(328): Error: Expecting an element , got nothing, at /rfc/middle/section[4]/section[2]/figure[1]
documents/antioch-v2.rfc.xml(358): Error: Expecting an element , got nothing, at /rfc/middle/section[4]/section[2]/figure[3]
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7990", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8140", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8140", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8140", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC2119", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8174", at None
documents/antioch-v2.rfc.xml(10): Error: Invalid document before running preptool.
Unable to complete processing documents/antioch-v2.rfc.xml
documents/antioch-v2.rfc.xml(103): Error: Expecting an element , got nothing, at /rfc/middle/section[3]/figure[1]
documents/antioch-v2.rfc.xml(140): Error: Expecting an element , got nothing, at /rfc/middle/section[3]/figure[3]
documents/antioch-v2.rfc.xml(209): Error: Expecting an element , got nothing, at /rfc/middle/section[4]/section[1]/figure[1]
documents/antioch-v2.rfc.xml(328): Error: Expecting an element , got nothing, at /rfc/middle/section[4]/section[2]/figure[1]
documents/antioch-v2.rfc.xml(358): Error: Expecting an element , got nothing, at /rfc/middle/section[4]/section[2]/figure[3]
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8140", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8140", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8140", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC2119", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7990", at None
documents/antioch-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC8174", at None
documents/antioch-v2.rfc.xml(10): Error: Invalid document before running preptool.
```

`example-v2.xml`:
```
documents/example-v2.rfc.xml(12): Error: Did not expect element area there, at /rfc/front/area
documents/example-v2.rfc.xml(94): Error: Expecting an element , got nothing, at /rfc/middle/section[1]/section/section[2]/figure[1]
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "ISO.IEC.10118-3", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "ISO.IEC.10118-3", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7253", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7253", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7253", at None
documents/example-v2.rfc.xml(8): Error: Invalid document before running preptool.
Unable to complete processing documents/example-v2.rfc.xml
documents/example-v2.rfc.xml(12): Error: Did not expect element area there, at /rfc/front/area
documents/example-v2.rfc.xml(94): Error: Expecting an element , got nothing, at /rfc/middle/section[1]/section/section[2]/figure[1]
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "ISO.IEC.10118-3", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "ISO.IEC.10118-3", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7253", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7253", at None
documents/example-v2.rfc.xml(0): Error: IDREF attribute target references an unknown ID "RFC7253", at None
documents/example-v2.rfc.xml(8): Error: Invalid document before running preptool.
```

opoudjis commented:
WE. NO. LONGER. SUPPORT. v2. That code is frozen. Are you *sure* you want to do this, @ronaldtse?

ronaldtse commented:
Oh, then let me comment them out. The V3 documents still need your attention, though.

ronaldtse committed:
Don’t process v2 sources

ronaldtse commented:
Updated RFC XML v3 problems:

antioch.xml:
```
echo "Running locally..."; bundle exec metanorma documents/antioch.xml
Running locally...
nil versions are discouraged and will be deprecated in Rubygems 4
documents/antioch.rfc.xml(12): Warning: Removed a redundant ascii= attribute from <title>
documents/antioch.rfc.xml(15): Warning: Removed a redundant asciiFullname= attribute from <author>
documents/antioch.rfc.xml(20): Error: Did not expect element style there, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[2]/*[1]
documents/antioch.rfc.xml(20): Error: Element defs has extra content: style, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[2]/*[1]
documents/antioch.rfc.xml(2630): Error: Did not expect element metadata there, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[3]
documents/antioch.rfc.xml(2630): Error: Element svg has extra content: metadata, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[3]
documents/antioch.rfc.xml(10): Warning: Invalid document after running preptool.
documents/antioch.rfc.xml(354): Warning: Found SVG with width or height specified, which will make the artwork not scale.  Specify a viewBox only to let the artwork scale.
 Created file documents/antioch.html
documents/antioch.rfc.xml(12): Warning: Removed a redundant ascii= attribute from <title>
documents/antioch.rfc.xml(15): Warning: Removed a redundant asciiFullname= attribute from <author>
documents/antioch.rfc.xml(20): Error: Did not expect element style there, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[2]/*[1]
documents/antioch.rfc.xml(20): Error: Element defs has extra content: style, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[2]/*[1]
documents/antioch.rfc.xml(2630): Error: Did not expect element metadata there, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[3]
documents/antioch.rfc.xml(2630): Error: Element svg has extra content: metadata, at /rfc/middle/section[4]/section[2]/figure[2]/artwork/*/*[3]
documents/antioch.rfc.xml(10): Warning: Invalid document after running preptool.
 Created file documents/antioch.txt
```

example.xml

```
echo "Running locally..."; bundle exec metanorma documents/example.xml
Running locally...
nil versions are discouraged and will be deprecated in Rubygems 4
documents/example.rfc.xml(8): Warning: Setting consensus="true" for IETF STD document (this is not the schema default, but is the only value permitted for this type of document)
documents/example.rfc.xml(11): Warning: Removed a redundant ascii= attribute from <title>
documents/example.rfc.xml(161): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(164): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(8): Warning: Expected a <link> with rel="prev" providing the datatracker url for the origin draft, or alternatively a "docName" attribute on <rfc> from which to construct such a <link> element.
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/boilerplate
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/toc
documents/example.rfc.xml(8): Warning: Invalid document after running preptool.
 Created file documents/example.html
documents/example.rfc.xml(8): Warning: Setting consensus="true" for IETF STD document (this is not the schema default, but is the only value permitted for this type of document)
documents/example.rfc.xml(11): Warning: Removed a redundant ascii= attribute from <title>
documents/example.rfc.xml(161): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(164): Warning: Removed a redundant ascii= attribute from <organization>
documents/example.rfc.xml(8): Warning: Expected a <link> with rel="prev" providing the datatracker url for the origin draft, or alternatively a "docName" attribute on <rfc> from which to construct such a <link> element.
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/boilerplate
documents/example.rfc.xml(0): Error: Expecting an element , got nothing, at /rfc/front/toc
documents/example.rfc.xml(8): Warning: Invalid document after running preptool.
 Created file documents/example.txt
```

opoudjis commented:
The antioch.xml problems are all because IETF are refusing to process SVG that exists in the real world, and insist on using a Mobile Phone profile of SVG from 2004 that noone can access from contemporary tools. I will not fix them.

In example.xml, they are complaining that I am not providing boilerplate. I won't, and I don't even know how to: that's the tool being silly. The ascii attributes probably are redundant, but they are legal, and inserted for illustration. I'll check the link issue, but nothing else looks to be legitimate.

opoudjis commented:
The  boilerplate is meant to be supplied by the "prep tool" described in https://tools.ietf.org/html/rfc7998. That's IETF's problem.

````
documents/example.rfc.xml(8): Warning: Expected a <link> with rel="prev" providing the datatracker url for the origin draft, or alternatively a "docName" attribute on <rfc> from which to construct such a <link> element.
````

I am adding the requested link element (whereas https://www.ietf.org/id/draft-levkowetz-xml2rfc-v3-implementation-notes-10.txt is actually requiring a redundant addition of docName in RFC documents to reference the draft). I could implement that behaviour, but given the mess v3 currently is, I'd rather sit and wait.

opoudjis commented:
xml2rfc 2.37 accepted SVG file, xml2rfc 2.42 is now crashing on it. I am going to either look for a random SVG now that doesn't crash xml2rfc, or omit completely.

opoudjis commented:
that's as much as I can do...
